apps/omg_status/lib/omg_status/monitor/memory_monitor.ex

```diff
@@ -0,0 +1,167 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Status.Monitor.MemoryMonitor do
+  @moduledoc """
+  Monitors and raises the :system_memory_too_high alarm when the system memory reaches
+  the specified threshold.
+
+  Intentionally raising a different alarm name from :memsup (which uses :system_memory_high_watermark)
+  so there is no ambiguity to which module is responsible for which alarm.
+
+  See http://erlang.org/pipermail/erlang-questions/2006-September/023144.html
+  """
+  use GenServer
+  require Logger
+
+  @type t :: %__MODULE__{
+          alarm_module: module(),
+          memsup_module: module(),
+          interval_ms: pos_integer(),
+          threshold: float(),
+          raised: boolean(),
+          tref: reference() | nil
```

Comment:
More meaningful name here would be good.

Reply:
Done! https://github.com/omisego/elixir-omg/pull/1474/commits/fc29514f8bb1fb907d527b56f7ca49af56dd7610